packages/shared/src/types/config/index.ts

```diff
@@ -97,6 +97,18 @@ export interface RsbuildConfig extends EnvironmentConfig {
   _privateMeta?: RsbuildConfigMeta;
 }
 
+export type MergedEnvironmentConfig = {
```

Comment:
I think the only different of `MergedEnvironmentConfig` and `NormalizedEnvironmentConfig` is `DeepReadonly<>`?

Can we just merge the two types? We do not want the config types to become so complex.

Reply:
`MergedEnvironmentConfig` only has configs that the user can modify, but `NormalizedEnvironmentConfig` has all configurations that are read-only.


`NormalizedEnvironmentConfig` = DeepReadonly<`MergedEnvironmentConfig`  + `SharedRsbuildConfig` (server / _privateMeta / ... )>